File: corporate-reporting/src/types/RiskUpdate.ts
import { IProgressUpdate, ProgressUpdate } from "./ProgressUpdate";
import { IEntity } from "./Entity";
import { ISignOff } from "./SignOff";
import { IHyperlink } from "./Hyperlink";

export interface IRiskUpdate extends IProgressUpdate {
    RiskID: number;
    RiskProbabilityID: number;
    RiskImpactLevelID: number;
    Escalate: boolean;
    EscalateToRiskRegisterID: number;
    DeEscalate: boolean;
    RiskMitigationActionUpdates: string;
    SendNotifications: boolean;
    RiskRegisterID: number;
    RiskProximity: Date;
    RiskCode: string;
    RiskIsOngoing: boolean;
    SignOffID: number;
    RiskAppetiteBreachAuthorised: boolean;
    RiskProbability?: IEntity;
    RiskImpactLevel?: IEntity;
    RiskRegister?: IEntity;
    SignOff?: ISignOff;
    Narrative: string;
    ClosureReason: string;
    Attachments?: IHyperlink[];
    ToBeDiscussed?: boolean;
    DiscussionForum?: string[];
}

export class RiskUpdate extends ProgressUpdate implements IRiskUpdate {
    public RiskID = null;
    public RiskProbabilityID = null;
    public RiskImpactLevelID = null;
    public Escalate = false;
    public EscalateToRiskRegisterID = null;
    public DeEscalate = false;
    public RiskMitigationActionUpdates = null;
    public SendNotifications = false;
    public RiskRegisterID = null;
    public RiskProximity = null;
    public RiskCode = '';
    public RiskIsOngoing = null;
    public SignOffID = null;
    public RiskAppetiteBreachAuthorised = false;
    public RiskProbability = null;
    public RiskImpactLevel = null;
    public Narrative = '';
    public ClosureReason = '';
    public UpdateUser = null;
    public Attachments = [];
    public ToBeDiscussed = false;
    public DiscussionForum = [];

    constructor(riskId?: number, period?: Date) {
        super(period);
        this.RiskID = riskId;
    }
}

File: ORB.Core/Models/RiskUpdate.cs
using System;
using System.Collections.Generic;

namespace ORB.Core.Models
{
    public partial class RiskUpdate : EntityUpdate
    {
        public string Comment { get; set; }
        public int? RagOptionID { get; set; }
        public int? RiskID { get; set; }
        public int? RiskProbabilityID { get; set; }
        public int? RiskImpactLevelID { get; set; }
        public bool? Escalate { get; set; }
        public int? EscalateToRiskRegisterID { get; set; }
        public bool? DeEscalate { get; set; }
        public bool? ToBeClosed { get; set; }
        public DateTime? UpdatePeriod { get; set; }
        public string RiskMitigationActionUpdates { get; set; }
        public bool? IsCurrent { get; set; }
        public bool? SendNotifications { get; set; }
        public int? RiskRegisterID { get; set; }
        public DateTime? RiskProximity { get; set; }
        public string RiskCode { get; set; }
        public bool? RiskIsOngoing { get; set; }
        public bool? RiskAppetiteBreachAuthorised { get; set; }
        public string Narrative { get; set; }
        public string ClosureReason { get; set; }
        public ICollection<Hyperlink> Attachments { get; set; }
        public bool? ToBeDiscussed { get; set; }
        public ICollection<string> DiscussionForum { get; set; }

        public virtual RagOption RagOption { get; set; }
        public virtual RiskImpactLevel RiskImpactLevel { get; set; }
        public virtual RiskProbability RiskProbability { get; set; }
        public virtual RiskRegister EscalateToRiskRegister { get; set; }
        public virtual RiskRegister RiskRegister { get; set; }
    }
}
